### PR TITLE
changed endpoint for getting al hightlights. Status code is now 200 w…

### PR DESCRIPTION
**Pull request by niraymak**
…hen the list is emtpy.

## Description

A few sentences describing the overall goals of the pull request's commits.

## Type of change

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I did not update API Controllers, if I did, I added/changed Postman tests
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes
-   [x] I updated the changelog with an end-user readable description
-   [x] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

1. Make sure there are no highlights in the database
2. Do a GET request to /api/Highlights (with right credentials)
3. Check if you receive an empty array with status code 200.

## Link to issue

Closes: #207

### CODE REVIEW

**Comment by Brend-Smits** [comment]
Hey @niraymak are there other endpoints from other controllers that will also need changing? I wonder if this is the only endpoint with this issue.

**Comment by niraymak** [comment]
> Hey @niraymak are there other endpoints from other controllers that will also need changing? I wonder if this is the only endpoint with this issue.

Following endpoint should be changed:

- GetAllEmbeddedProjects also sends 404.
- GetAllRoles sends 404 if list.count == 0.
- GetAllPossibleScopes sends 404 if list.count == 0.

Shall I change them and push them to this PR?

**Comment by niraymak** [comment]
> > Hey @niraymak are there other endpoints from other controllers that will also need changing? I wonder if this is the only endpoint with this issue.
> 
> Following endpoint should be changed:
> 
> * GetAllEmbeddedProjects also sends 404.
> * GetAllRoles sends 404 if list.count == 0.
> * GetAllPossibleScopes sends 404 if list.count == 0.
> 
> Shall I change them and push them to this PR?

Changed the endpoints and pushed them to the PR

**Comment by Brend-Smits** [comment]
Awesome, thanks @niraymak. Please also change the postman tests ^^'

**Comment by niraymak** [comment]
@Brend-Smits test results are 100%. The corresponding tests already check for status code 200. I am not sure what to update in the Postman tests. Can you tell me what tests I should add?

**Comment by Brend-Smits** [comment]
> @Brend-Smits test results are 100%. The corresponding tests already check for status code 200. I am not sure what to update in the Postman tests. Can you tell me what tests I should add?

Ah right, we don't have any tests for checking status codes when there are no entries. That's fine.

**Review by Brend-Smits — CHANGES_REQUESTED** [review]
Remove try-catch as discussed in Teams and we're good to go.

File: API/Controllers/RoleController.cs [hunk 1]
```diff
@@ -69,17 +61,22 @@ public async Task<IActionResult> GetAllRoles()
         [Authorize(Policy = nameof(Scopes.RoleRead))]
         public IActionResult GetAllPossibleScopes()
         {
-            List<string> scopeList = roleService.GetValidScopes();
-            if(scopeList.Count == 0)
+            List<string> scopeList;
```
Comment (Brend-Smits) [inline API/Controllers/RoleController.cs:64]: I think this section should be reverted to the way it was before.

**Review by Brend-Smits — CHANGES_REQUESTED** [review]

**Comment by Brend-Smits** [comment]
Since #240 is now merged, make sure the documentation is updated!

**Review by Brend-Smits — APPROVED** [review]

**Pull request merged**